### ISSUE TITLE
Chatterino2 on Fedora - floating point exception while connecting(?) to chat

### ISSUE THREAD

**Issue by Soveu** [Chatterino/chatterino2#2565]
**Describe the bug**

When joining a chat, program generates floating point exception

**To reproduce**

Join the chat and wait for a message

**Screenshots**

**Chatterino version**

2.2.2

**Operating system**

Fedora 33

**Additional information**

backtrace from crashed thread
```
0: std::__detail::_Map_base<chatterino::ColorType, std::pair<chatterino::ColorType const, std::shared_ptr<QColor> >, std::allocator<std::pair<chatterino::ColorType const, std::shared_ptr<QColor> > >, std::__detail::_Select1st, std::equal_to<chatterino::ColorType>, std::hash<chatterino::ColorType>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true>, true>::at(chatterino::ColorType const&)
1: chatterino::ColorProvider::color(chatterino::ColorType) const
2: chatterino::SharedMessageBuilder::parseHighlights()
3: chatterino::TwitchMessageBuilder::build()
4: chatterino::Outcome chatterino::TwitchChannel::loadRecentMessages()::{lambda(auto:1)#1}::operator()<chatterino::NetworkResult>(chatterino::NetworkResult) const
5: std::_Function_handler<chatterino::Outcome (chatterino::NetworkResult), chatterino::TwitchChannel::loadRecentMessages()::{lambda(auto:1)#1}>::_M_invoke(std::_Any_data const&, chatterino::NetworkResult&&) [clone .lto_priv.0]
6: QtConcurrent::StoredFunctorCall0<void, chatterino::loadUncached(std::shared_ptr<chatterino::NetworkData> const&)::{lambda()#1}::operator()()::{lambda()#3}::operator()()::{lambda()#2}>::runFunctor()
7: non-virtual thunk to QtConcurrent::RunFunctionTask<void>::run()
```

**Comment by leon-richardt** [comment]
Pretty much identical to https://github.com/Chatterino/chatterino2/issues/2355, with the difference that the reporter compiled Chatterino themselves (notably with `-flto`).

How did you install Chatterino on your system? Compiled yourself, AppImage, distro package manager?

**Comment by Soveu** [comment]
installed via `dnf`

**Comment by ilyazzz** [comment]
It's an issue with fedora's package of chatterino, install from source or use the appimage instead.

**Comment by Frankstar** [comment]
@ilyazzz 
well its a workaround - ty for that.
its stated on the offical webpage we should use `dnf install chatterino2` on Fedora.

you should remove it from there or fix it - as currently i would flag it as broken.

**Comment by leon-richardt** [comment]
> you should remove it from there or fix it - as currently i would flag it as broken.

If it actually is a problem with the Fedora package (which it seems to be when a source-install works fine), that would be the task for the package maintainer, not the upstream developers.

Have you reported this issue on the Fedora bug tracker already?

**Comment by Frankstar** [comment]
i get you.
still its on YOUR documentation as the way to go.
no i didnt report it yet.
you know your maintainer?

**Comment by leon-richardt** [comment]
I was referring to the "fix it" part.

Not sure what you mean by "our" maintainer but the Fedora package maintainer is listed on the official [Fedora package page](https://src.fedoraproject.org/rpms/chatterino2).

**Comment by ilyazzz** [comment]
@fourtf could you also add a link to https://flathub.org/apps/details/com.chatterino.chatterino on that page?

**Comment by Soveu** [comment]
It is fixed with 2.3.1 on Fedora 34

**Comment by zneix** [comment]
> It is fixed with 2.3.1 on Fedora 34

If reoccurs, feel free to comment below and reopen :+1:
